pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/AssemblyRenderServlet.java

```diff
@@ -0,0 +1,121 @@
+ * with "ctx_".
+ *
+ * For example, if an asciidoc attribute of name 'product' needs to be passed, there will need to be a
+ * query parameter of name 'ctx_product' provided in the url.
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet which transforms asciidoc content into html",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletResourceTypes(
+        resourceTypes = { "pantheon/assembly" },
+        methods = "GET",
+        extensions = "preview")
+@SuppressWarnings("serial")
+public class AssemblyRenderServlet extends SlingSafeMethodsServlet {
+
+    private final Logger log = LoggerFactory.getLogger(AssemblyRenderServlet.class);
+
+    static final String PARAM_RERENDER = "rerender";
+    static final String PARAM_DRAFT = "draft";
+    static final String PARAM_LOCALE = "locale";
+    static final String PARAM_VARIANT = "variant";
+
+    private AsciidoctorService asciidoctorService;
+
+    @Activate
+    public AssemblyRenderServlet(
+            @Reference AsciidoctorService asciidoctorService) {
+        this.asciidoctorService = asciidoctorService;
+    }
+
+    @Override
+    public void doGet(SlingHttpServletRequest request,
+            SlingHttpServletResponse response) throws IOException {
+        String locale = paramValue(request, PARAM_LOCALE, DEFAULT_MODULE_LOCALE.toString());
+        boolean draft = paramValueAsBoolean(request, PARAM_DRAFT);
+        boolean reRender = paramValueAsBoolean(request, PARAM_RERENDER);
+        String variantName = paramValue(request, PARAM_VARIANT, DEFAULT_VARIANT_NAME);
+
+        Module module = request.getResource().adaptTo(Module.class);
+        Locale localeObj = LocaleUtils.toLocale(locale);
+
+        Optional<HashableFileResource> moduleVariantSource = null;
+
+        switch(paramValue(request, PARAM_DRAFT)){
+            case "true":
+                moduleVariantSource = module.moduleLocale(localeObj)
+                        .traverse()
+                        .toChild(ModuleLocale::source)
+                        .toChild(SourceContent::draft)
+                        .getAsOptional();
+                break;
+
+            case "false":
+                moduleVariantSource = module.moduleLocale(localeObj)
+                        .traverse()
+                        .toChild(ModuleLocale::source)
+                        .toChild(SourceContent::released)
+                        .getAsOptional();
+                break;
+        }
+
+
+        if(!moduleVariantSource.isPresent()) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, (draft ? "Draft " : "Released ")
+                    + "source content not found for " + variantName +  " module variant at "
+                    + request.getResource().getPath());
+        }
```

Comment:
```suggestion
        if(!moduleVariantSource.isPresent()) {
            response.sendError(HttpServletResponse.SC_NOT_FOUND, (draft ? "Draft " : "Released ")
                    + "source content not found for " + variantName +  " module variant at "
                    + request.getResource().getPath());
            return;
        }
```

Reply:
I have a query, here as we are using sendError method, shoudn't it automatically returns control back to parent class?
     * After using this method, the response should be considered
     * to be committed and should not be written to.

Reply:
Hi @rednitish, the answer to your question is no, Java does not work that way. There is no way to call a method, and then that child method halts further execution of its parent method, outside of throwing an exception (which is not what's happening here). You must add an explicit `return` in order to stop your method from continuing to execute once the child finishes.

What the javadoc is saying is that once you call response.sendError, you should not attempt to call any further methods on the response object, because they will fail.